cpp/src/arrow/util/bpacking.cc

```diff
@@ -29,6 +29,9 @@
 #if defined(ARROW_HAVE_RUNTIME_AVX512)
 #include "arrow/util/bpacking_avx512.h"
 #endif
+#if defined(ARROW_HAVE_AVX512_ICX)
```

Comment:
```suggestion
#if defined(ARROW_HAVE_RUNTIME_AVX512_ICX)
```

We need to define `ARROW_HAVE_RUNTIME_AVX512_ICX` in `cpp/cmake_modules/SetupCxxFlags.cmake`.

Reply:
@kou The code has been changed.